sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -1099,21 +1066,83 @@ case class UnsetTableProperties(
     copy(table = newChild)
 }
 
-trait AlterTableCommand extends UnaryCommand {
+trait AlterTableColumnCommand extends UnaryCommand {
   def table: LogicalPlan
-  def operation: String
   def changes: Seq[TableChange]
   override def child: LogicalPlan = table
 }
 
+/**
+ * The logical plan of the ALTER TABLE ... ADD COLUMNS command.
+ */
+case class AlterTableAddColumns(
+    table: LogicalPlan,
+    columnsToAdd: Seq[QualifiedColType]) extends AlterTableColumnCommand {
+  import org.apache.spark.sql.connector.catalog.CatalogV2Util._
+  columnsToAdd.foreach { c =>
+    failNullType(c.dataType)
+    TypeUtils.failWithIntervalType(c.dataType)
+  }
+
+  override def changes: Seq[TableChange] = {
+    columnsToAdd.map { col =>
+      require(col.position.forall(_.resolved),
```

Comment:
do we need the `path` to be resolved as well?

Reply:
Updated.